packages/statemanager/src/ethersStateManager.ts

```diff
@@ -0,0 +1,420 @@
+import { blockFromRpc } from '@ethereumjs/block/dist/from-rpc'
+import { RLP } from '@ethereumjs/rlp'
+import { Trie } from '@ethereumjs/trie'
+import {
+  Account,
+  arrToBufArr,
+  bigIntToHex,
+  bufferToHex,
+  intToHex,
+  isHexPrefixed,
+  toBuffer,
+} from '@ethereumjs/util'
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { debug } from 'debug'
+import { hexToBytes } from 'ethereum-cryptography/utils'
+
+import { Cache } from './cache'
+
+import { BaseStateManager } from '.'
+
+import type { Proof, StateManager } from '.'
+import type { StorageDump } from './interface'
+import type { StorageProof } from './stateManager'
+import type { Common } from '@ethereumjs/common'
+import type { Address, PrefixedHexString } from '@ethereumjs/util'
+
+const log = debug('statemanager')
+export interface EthersStateManagerOpts {
+  provider: string | JsonRpcProvider
+  blockTag?: bigint | 'latest' | 'earliest'
+}
+
+export class EthersStateManager extends BaseStateManager implements StateManager {
+  private provider: JsonRpcProvider
+  private contractCache: Map<string, Buffer>
+  private storageTries: { [key: string]: Trie }
+  private externallyRetrievedStorageKeys: Map<string, Map<string, boolean>>
+  private blockTag: string
+  private trie: Trie
+
+  constructor(opts: EthersStateManagerOpts) {
+    super({})
+    this.trie = new Trie({ useKeyHashing: true })
```

Comment:
```suggestion
    // useKeyHashing = true since the web3 api provides proof nodes which are hashed
    // If there were direct api access to devp2p stack, a normal Trie could have been constructed
    this.trie = new Trie({ useKeyHashing: true })
```